src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/NewModuleDialogValidator.java

```diff
@@ -1,204 +0,0 @@
-        final String packageName = dialog.getPackageName();
-        if (packageName.length() == 0) {
-            final String errorMessage = validatorBundle.message(
-                    NOT_EMPTY,
-                    "Package Name"
-            );
-            JOptionPane.showMessageDialog(
-                    null,
-                    errorMessage,
-                    errorTitle,
-                    JOptionPane.ERROR_MESSAGE
-            );
-
-            return false;
-        }
-
-        if (!packageName.matches(RegExUtil.ALPHANUMERIC)) {
-            final String errorMessage = validatorBundle.message(
-                    "validator.alphaNumericCharacters",
-                    "Package Name"
-            );
-            JOptionPane.showMessageDialog(
-                    null,
-                    errorMessage,
-                    errorTitle,
-                    JOptionPane.ERROR_MESSAGE
-            );
-
-            return false;
-        }
-
-        if (!Character.isUpperCase(packageName.charAt(0))
-                && !Character.isDigit(packageName.charAt(0))
-        ) {
-            final String errorMessage = validatorBundle.message(
-                    "validator.startWithNumberOrCapitalLetter",
-                    "Package Name"
-            );
-            JOptionPane.showMessageDialog(
-                    null,
-                    errorMessage,
-                    errorTitle,
-                    JOptionPane.ERROR_MESSAGE
-            );
-
-            return false;
-        }
-
-        final String moduleName = dialog.getModuleName();
-        if (moduleName.length() == 0) {
-            final String errorMessage = validatorBundle.message(
-                    NOT_EMPTY,
-                    MODULE_NAME
-            );
-            JOptionPane.showMessageDialog(
-                    null,
-                    errorMessage,
-                    errorTitle,
-                    JOptionPane.ERROR_MESSAGE
-            );
-
-            return false;
-        }
-
-        if (!moduleName.matches(RegExUtil.ALPHANUMERIC)) {
-            final String errorMessage = validatorBundle.message(
-                    "validator.alphaNumericCharacters",
-                    MODULE_NAME
-            );
-            JOptionPane.showMessageDialog(
-                    null,
-                    errorMessage,
-                    errorTitle,
-                    JOptionPane.ERROR_MESSAGE
-            );
-
-            return false;
-        }
-
-        if (moduleName.equals(packageName)) {
-            final String errorMessage = validatorBundle.message(
-                    "validator.moduleNameIsTheSameAsPackage",
-                    MODULE_NAME
-            );
-            JOptionPane.showMessageDialog(
-                    null,
-                    errorMessage,
-                    errorTitle,
-                    JOptionPane.ERROR_MESSAGE
-            );
-
-            return false;
-        }
```

Comment:
This exists to check if the module name and package name are identical and throws an error if so. 
- Do we need this check? 
- How do we perform this through annotations? Module name field validation needs to access the package name to compare.

I'm unsure on how to proceed with this. Please advise. Thanks!

Reply:
There no system restrictions to call module e.g `Vendor_Vendor`.
However, this is wrong from the best practices standpoint.

@bohdan-harniuk 
How do you think it can be implemented?

Reply:
I understand @VitaliyBoyko, but just to note: a few modules which I noticed that have the same vendor and module name
- elasticsearch/elasticsearch
- container-interop/container-interop
- cache/cache
- mockery/mockery